confidential-data-hub/hub/src/bin/config/mod.rs

```diff
@@ -65,13 +75,31 @@ impl Default for CdhConfig {
 }
 
 impl CdhConfig {
-    pub async fn init(config_path: &str) -> Result<Self> {
-        let mut config = Self::from_file(config_path).unwrap_or_else(|e| {
-            warn!("read config file {config_path} failed {e:?}, use a default config where `aa_kbc_params` = offline_fs_kbc::null.");
-            Self::default()
+    pub fn new(config_path: Option<String>) -> Result<Self> {
+        let config_path = config_path.or_else(|| {
+            if let std::result::Result::Ok(env_path) = env::var("CDH_CONFIG_PATH") {
```

Comment:
nit: can we move this one layer up to the main() fn invoking `CdhConfig::new()`? It might unclutter the logic here. We already log which config file we are using below ("Use configuration file...").

```rust
let cfg_path_env = env::var("CDH_CONFIG_PATH").ok();
let cfg_path = cli.config.or(cfg_path_env);
let config = CdhConfig::new(cfg_path)?;
```